lib/gui/app.js

```diff
@@ -249,13 +252,17 @@ app.run(($window) => {
       description: messages.warning.exitWhileFlashing()
     }).then((confirmed) => {
       if (confirmed) {
+        debug('aborting')
         analytics.logEvent('Close confirmed while flashing', {
           uuid: flashState.getFlashUuid()
         })
 
-        // This circumvents the 'beforeunload' event unlike
-        // electron.remote.app.quit() which does not.
-        electron.remote.process.exit(EXIT_CODES.SUCCESS)
+        childWriter.abort(() => {
+          // This circumvents the 'beforeunload' event unlike
+          // electron.remote.app.quit() which does not.
+          debug('aborted')
+          electron.remote.app.exit(EXIT_CODES.SUCCESS)
+        })
       }
 
       analytics.logEvent('Close rejected while flashing')
```

Comment:
If `confirmed` is true, then chances are that this message will be shown (and sent to Mixpanel) before the application exits. Looks like we should wrap `.abort()` in a promise.

Reply:
A simple `else` branch does the trick just as well :)